app/services/backup_code_generator.rb

```diff
@@ -12,16 +12,15 @@ def initialize(user, num_words: BackupCodeConfiguration::NUM_WORDS)
     @user = user
   end
 
-  # @return [Array<String>]
-  def generate
-    delete_existing_codes
-    generate_new_codes
-  end
+  def delete_and_regenerate(salt: SecureRandom.hex(32))
+    codes = generate_new_codes
+
+    BackupCodeConfiguration.transaction do
+      @user.backup_code_configurations.destroy_all
```

Comment:
Is this potentially adding a new query where one wouldn't have existed previously in the pure create-from-none use-case? Is that a concern?

Reply:
It does potentially, but I think since we'd probably want to verify whether there are any previously existing with a database query before, we may as well just delete.